test/unit/videoSource.test.js

```diff
@@ -297,3 +298,66 @@ describe('tests withCredentials', () => {
 });
 
 
+describe('test isCodecAlreadyExist method', () => {
```

Comment:
i'd also add tests for the generated url, to see that codec isn't added or is added when needed

Reply:
@maoznir done